backend/climateconnect_api/views/user_views.py

```diff
@@ -95,15 +95,16 @@ def get(self, request):
 
 
 class MemberProfilesView(ListAPIView):
-    permission_classes = (IsAuthenticated,)
-    serializer_class = UserProfileSerializer
+    permission_classes = [AllowAny]
     pagination_class = PageNumberPagination
     filter_backends = [SearchFilter]
     search_fields = ['url_slug']
 
-    def get_queryset(self):
-        if not UserProfile.objects.filter(user=self.request.user).exists() or \
-                not self.request.user.user_profile.is_profile_verified:
-            raise PermissionDenied(detail="You do not have permission to access this page.")
-
-        return UserProfile.objects.filter(is_profile_verified=True)
+    def get_queryset(self, request):
+        user_profiles = UserProfile.objects.filter(is_profile_verified=True)
+        if request.user.is_authenticated(): 
+            serializer = UserProfileSerializer(user_profiles)
+            return Response(serializer.data, status=status.HTTP_200_OK)
```

Comment:
It will be self.request.user. Django rest generics API views include request param inside main class